core/src/test/scala/unit/kafka/server/LogRecoveryTest.scala

```diff
@@ -212,7 +228,7 @@ class LogRecoveryTest extends QuorumTestHarness {
     server2.startup()
     updateProducer()
     // check if leader moves to the other server
-    leader = waitUntilLeaderIsElectedOrChanged(zkClient, topic, partitionId, oldLeaderOpt = Some(leader))
+    leader = awaitLeaderChange(servers, topicPartition, leader)
```

Comment:
I think waitUntilLeaderIsElectedOrChangedWithAdmin can use to instead of it.

Reply:
Is there a reason `awaitLeaderChange` cannot be used? It seems to be much quicker than `waitUntilLeaderIsElectedOrChangedWithAdmin` which does describeTopics().

Reply:
I took quick look at code, and i think you are right.